### PR TITLE
draft: feat(frontend): add ImportStatusPropertyField Component

### PR DESCRIPTION
**Pull request by irumvanselme**

### CODE REVIEW

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx [hunk 1]
```diff
@@ -22,8 +29,91 @@ export const DATA_TEST_ID = {
...
+          </Typography>
+      )
+    }
+
+    if((status === ImportProcessStateEnums.Enums.Status.COMPLETED) && !result.errored && !result.parsingErrors && result.parsingWarnings) {
+      return (
+        <Typography data-testid={DATA_TEST_ID.MESSAGE}>
+          Completed with
+          <Box component={"span"} fontWeight={"bold"}> parsing warnings</Box>
+          <HelpTip>
```
Comment (jacob-js) [inline frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx:77]: A switch statement is needed here. The way we do with the ImportProcessStateIcon.

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx [hunk 2]
```diff
@@ -22,8 +29,91 @@ export const DATA_TEST_ID = {
...
+        </Typography>
+      )
+    }
+
+    return <Typography>Unknown Status</Typography>
+  }
+
+  return (
+    <PropertyFieldLayout title={"Status"} fieldId={props.fieldId}>
+      <Box display={"flex"} gap={theme.spacing(1)} alignItems={"center"} data-testid={DATA_TEST_ID.IMPORT_STATUS_FIELD}>
```
Comment (jacob-js) [inline frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx:111]: For the format, please do not enclose strings in curly-braces.

Comment (jacob-js) [inline frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx:111]: For the format, please do not enclose strings in curly braces.

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx [hunk 3]
```diff
@@ -0,0 +1,215 @@
...
+import React from "react";
+import { v4 as uuid } from "uuid"
+import "@testing-library/jest-dom";
+import { render, screen } from "src/_test_utilities/test-utils";
+import * as States from "./ImportStatusPropertyField.stories";
+import ImportProcessStateEnums from "api-specifications/importProcessState";
+import { DATA_TEST_ID as HELP_TIP_DATA_TEST_ID } from "src/theme/HelpTip/HelpTip";
+import ImportStatusPropertyField, { DATA_TEST_ID } from "./ImportStatusPropertyField";
+import { DATA_TEST_ID as ICON_DATA_TEST_ID } from "src/modeldirectory/components/ImportProcessStateIcon/ImportProcessStateIcon";
+import { ModelInfoTypes } from "../../../../../../../../../../modelInfo/modelInfoTypes";
```
Comment (jacob-js) [inline frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx:13]: Use the absolute path for the import

**Review by jacob-js — CHANGES_REQUESTED** [review]

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx [hunk 4]
```diff
@@ -0,0 +1,215 @@
...
+    expect(console.error).not.toHaveBeenCalled();
+    expect(console.warn).not.toHaveBeenCalled();
+
+    expect(screen.getByTestId(DATA_TEST_ID.IMPORT_STATUS_FIELD)).toMatchSnapshot()
+  })
+
+  it("Should Render correctly for Running State", () => {
+    // GIVEN the ImportStatusPropertyField component is rendered with running import state.
+    const importProcessState = States.Running.args?.importProcessState!;
+
```
Comment (jacob-js) [inline frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx:54]: These tests can be made shorter by writing them in a parameterized way. Please refer to the tests for the importProcessStateIcon for example.

**Review by jacob-js — CHANGES_REQUESTED** [review]

**Review by rav3n11 — COMMENTED** [review]
@irumvanselme I've added some changes based off of this branch and opened a pull request  [here](https://github.com/tabiya-tech/taxonomy-model-application/pull/311)... Check it out?

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx (hunk 3 above)
Comment (irumvanselme) [reply frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.test.tsx]: Fixing it

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx (hunk 2 above)
Comment (irumvanselme) [reply frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx]: Fixing it

Comment (irumvanselme) [reply frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx]: Fixing it

File: frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx (hunk 1 above)
Comment (irumvanselme) [reply frontend/src/modeldirectory/components/ModelProperties/components/ModelPropertiesContent/components/ModelPropertiesImportExport/ImportTimeline/components/ImportStatusPropertyField/ImportStatusPropertyField.tsx]: Fixing it

**Pull request closed without merging**